src/mappings/shared/handleTokenEntity.ts

```diff
@@ -1,44 +1,166 @@
-import { create, getOptional } from '@kodadot1/metasquid/entity'
+import { create, getOptional, getWith } from '@kodadot1/metasquid/entity'
 import md5 from 'md5'
 import { CollectionEntity as CE, NFTEntity as NE, TokenEntity as TE } from '../../model'
-import { warn } from '../utils/logger'
+import { debug, warn } from '../utils/logger'
 import { Context } from '../utils/types'
 
 const OPERATION = 'TokenEntity' as any
 
-export async function handleTokenEntity(context: Context, collection: CE, nft: NE): Promise<TE | undefined> {
-  const nftMedia = nft.image || nft.media
+function generateTokenId(collectionId: string, nftMedia: string): string {
+  return `${collectionId}-${md5(nftMedia)}`
+}
+
+async function createToken(context: Context, collection: CE, nft: NE): Promise<TE | undefined> {
+  const nftMedia = nft.image ?? nft.media
   if (!nftMedia || nftMedia === '') {
     warn(OPERATION, `MISSING NFT MEDIA ${nft.id}`)
     return
   }
+  const tokenId = generateTokenId(collection.id, nftMedia)
+  debug(OPERATION, { createToken: `Create TOKEN ${tokenId} for NFT ${nft.id}` })
+  const tokenName = typeof nft.name === 'string' ? nft.name?.replace(/([#_]\d+$)/g, '').trim() : ''
 
-  const tokenId = `${collection.id}-${md5(nftMedia)}`
-  let token = await getOptional<TE>(context.store, TE, tokenId)
+  const token = create(TE, tokenId, {
+    createdAt: nft.createdAt,
+    collection,
+    name: tokenName,
+    count: 1,
+    totalCount: 1,
+    hash: md5(tokenId),
+    image: nft.image,
+    media: nft.media,
+    metadata: nft.metadata,
+    meta: nft.meta,
+    blockNumber: nft.blockNumber,
+    updatedAt: nft.updatedAt,
+    id: tokenId,
+  })
 
-  if (!token) {
-    const tokenName = (typeof nft.name === 'string' ? nft.name?.replace(/([#_]\d+$)/g, '').trim() : '')
-
-    token = create(TE, tokenId, {
-      createdAt: nft.createdAt,
-      collection,
-      name: tokenName,
-      count: 1,
-      hash: md5(tokenId),
-      image: nft.image,
-      media: nft.media,
-      blockNumber: nft.blockNumber,
-      updatedAt: nft.updatedAt,
-      id: tokenId,
-    })
-  } else {
-    token.count += 1
-  }
+  nft.token = token
+  await context.store.save(token)
+  await context.store.save(nft)
```

Comment:
any particular reason to `save` nft here? 
cause slowest op is

Reply:
it needs to be updated with new token
but indeed, update is more suitable and faster